src/main/java/org/jabref/logic/bibtexkeypattern/BracketedPattern.java

```diff
@@ -106,40 +106,68 @@ public static String expandBrackets(String pattern, Character keywordDelimiter,
         Objects.requireNonNull(pattern);
         Objects.requireNonNull(entry);
         StringBuilder sb = new StringBuilder();
-        StringTokenizer st = new StringTokenizer(pattern, "\\[]", true);
+        StringTokenizer st = new StringTokenizer(pattern, "\\[]\"", true);
 
         while (st.hasMoreTokens()) {
             String token = st.nextToken();
-            if ("\\".equals(token)) {
-                if (st.hasMoreTokens()) {
-                    sb.append(st.nextToken());
-                }
-                // FIXME: else -> raise exception or log? (S.G.)
-            } else {
-                if ("[".equals(token)) {
-                    // Fetch the next token after the '[':
+            if ("\"".equals(token)) {
+                sb.append(token);
+                while (st.hasMoreTokens()) {
                     token = st.nextToken();
-                    List<String> fieldParts = parseFieldMarker(token);
-                    // check whether there is a modifier on the end such as
-                    // ":lower":
-                    if (fieldParts.size() <= 1) {
-                        sb.append(getFieldValue(entry, token, keywordDelimiter, database));
-                    } else {
-                        // apply modifiers:
-                        String fieldValue = getFieldValue(entry, fieldParts.get(0), keywordDelimiter, database);
-                        sb.append(applyModifiers(fieldValue, fieldParts, 1));
+                    sb.append(token);
+                    if ("\"".equals(token)) {
+                        break;
                     }
-                    // Fetch and discard the closing ']'
+                }
+            } else {
+                if ("\\".equals(token)) {
                     if (st.hasMoreTokens()) {
+                        sb.append(st.nextToken());
+                    }
+                    // FIXME: else -> raise exception or log? (S.G.)
+                } else {
+                    if ("[".equals(token)) {
+                        // Fetch the next token after the '[':
                         token = st.nextToken();
+                        Boolean foundClosingBracket = false;
+                        // make sure to read until the next ']'
+                        while (st.hasMoreTokens()) {
+                            String subtoken = st.nextToken();
```

Comment:
Question: you fetch the next token above in line 131, and then here the subtoken. So this moves the stream forward twice, right? What happens if the user adds an empty brace, i.e `[]`. Is the closing bracket still found? Can you please add a test for this case as well. Thanks!

Reply:
Good catch! But what should "[]" expand to? Inside a quote, the parsing should work fine. but what does "[]" mean in terms of JabRef fields? Should it just do nothing and print an appropriate warning?

Reply:
Yes, I would say it should do nothing and log a warning. I don't really see a use case for the empty bracket, it should just not break the parser.

Reply:
Done https://github.com/JabRef/jabref/pull/6469/commits/230fef2a16cd09ec3bd50647b7499ae943608e80